### DISCUSSION TITLE
Does superset supports dynamic chart loading wrt different entities in the table?

### DISCUSSION THREAD

**Discussion by jyotidhiman0610** [apache/superset#18285] [Q&A / Help]
I have a use case like below:
Input Data:

```
entity          test_run                some_values        
A                    x                         10                       ===> Chart 1 for entity A
A                    y                         20                       ===> Chart 2 for entity A
A                    z                         45                       ===> Chart 3 for entity A
B                    x                         56                       ===> Chart 1 for entity B 
B                    z                         89                       ===> Chart 2 for entity B
```

Dashboard should have
- 3 charts for entity A
- 2 charts for entity B

If a new entity comes in data, it should have n charts according to values in "test_run"
The number of distinct values in "test_run" are known and charts can be defined for each value of "test_run", like chart for test_run "x" will always be a pie chart, etc.

Any way to it dynamically?

Kindly let me know if more details are needed.

**Comment by zufolo441** [comment]
In my opinion, you have to write some web service that cycles the query result and generates an array of json of pie chart and passes it in URL to superset.The charts could be rendered in an array of <iframe/>. Here is an example for a pie chart:
https://localhost:8088/superset/explore/?form_data={"viz_type":"pie";"datasource":"113__table";"slice_id":512;"url_params":{};"time_range_endpoints":["inclusive";"exclusive"];"granularity_sqla":"data_nascita";"time_range":"No+filter";"groupby":["nome"];"metric":"count";"adhoc_filters":[{"clause":"WHERE";"comparator":"M";"expressionType":"SIMPLE";"filterOptionName":"filter_0qoax0rgoqyr_7dul8e6vj8j";"isExtra":false;"isNew":false;"operator":"==";"sqlExpression":null;"subject":"sesso"}];"row_limit":100;"sort_by_metric":true;"color_scheme":"supersetColors";"legendType":"scroll";"legendOrientation":"top";"label_type":"key_percent";"number_format":">;.2~f";"show_labels":true;"labels_outside":true;"outerRadius":70;"donut":true;"innerRadius":30;"extra_form_data":{"custom_form_data":{};"override_form_data":{};"append_form_data":{}}}
